src/quicktype-core/language/Java.ts

```diff
@@ -350,7 +362,11 @@ export class JavaRenderer extends ConvenienceRenderer {
 
     protected javaTypeWithoutGenerics(reference: boolean, t: Type): Sourcelike {
         if (t instanceof ArrayType) {
-            return [this.javaTypeWithoutGenerics(false, t.items), "[]"];
+            if (this._options.useList) {
+                return [this.javaTypeWithoutGenerics(true, t.items)];
```

Comment:
This seems weird.  Shouldn't this be `List`?

Reply:
Yeah you're right. I probably wrote it in [line 467](https://github.com/quicktype/quicktype/blob/47dd21307560cd515a306a40fb0eab03b224d2a5/src/quicktype-core/language/Java.ts#L467) because of the name of them method *javaTypeWithoutGenerics()*. `List<T>` is called a generic type in Java. But I will see what changes if I put it into the method. If everything is fine I can change it. Just to be sure, so you propose to change it to the following?
```ts
return ["List<", this.javaTypeWithoutGenerics(true, t.items), ">"]
```

Reply:
No, it seems that this should just be `List`, like it is for `Map` a few lines further down.  Maybe that doesn't work in Java?

Reply:
@schani Sorry for being a bit unresponsive. I am only able to work on this on weekends and unfortunately I crashed my local dev environment when trying to replace docker with podman. Your proposal to just return `List` doesn't work. E.g. in `Deserializer` code we'd end up with

```
jsonParser.readValueAs(new TypeReference<List>() {});
```
instead of
```
jsonParser.readValueAs(new TypeReference<List<Bar>>() {});
```
However, Jackson requires `Bar` to being able to infer the type of the elements of a list with the correct type. I am not sure why returning just `"Map"` works but in general there are a few differences between the array cases and the Map cases. Eventually I now changed the code to 

```java
return ["List<", this.javaTypeWithoutGenerics(true, t.items), ">"];
```
and would submit this as eventual PR from my side. I would leave any further modifications up to you or the community.